internal/service/cluster/cluster.go

```diff
@@ -246,11 +253,17 @@ func (s *Service) Run(ctx context.Context, host service.Host) error {
 
 	peers, err := s.getPeers()
 	if err != nil {
-		return fmt.Errorf("failed to get peers to join: %w", err)
+		// We can continue here as we have a rejoin goroutine try again later.
+		level.Warn(s.log).Log("msg", "failed to get peers to join at startup", "err", err)
+		peers = nil
 	}
 
-	level.Info(s.log).Log("msg", "starting cluster node", "peers", strings.Join(peers, ","),
-		"advertise_addr", s.opts.AdvertiseAddress)
+	level.Info(s.log).Log(
```

Comment:
That explanation makes sense; it just wasn't consistent so I wasn't sure if it was intentional. Maybe a small comment about why it's info level would be helpful to future readers?

Reply:
Cool, added a comment. Also worth noting that `getPeers` selects only `ClusterMaxJoinPeers`, but this can still be long cause users can still set it to `0` meaning return all the peers.